app/src/main/java/org/schabi/newpipe/CheckForNewAppVersion.java

```diff
@@ -118,104 +122,110 @@ private static String byte2HexFormatted(final byte[] arr) {
             final Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(apkLocationUrl));
             final PendingIntent pendingIntent
-                    = PendingIntent.getActivity(APP, 0, intent, 0);
+                    = PendingIntent.getActivity(application, 0, intent, 0);
 
-            final NotificationCompat.Builder notificationBuilder = new NotificationCompat
-                    .Builder(APP, APP.getString(R.string.app_update_notification_channel_id))
+            final String channelId = application
+                    .getString(R.string.app_update_notification_channel_id);
+            final NotificationCompat.Builder notificationBuilder
+                    = new NotificationCompat.Builder(application, channelId)
                     .setSmallIcon(R.drawable.ic_newpipe_update)
                     .setVisibility(NotificationCompat.VISIBILITY_PUBLIC)
                     .setContentIntent(pendingIntent)
                     .setAutoCancel(true)
-                    .setContentTitle(APP.getString(R.string.app_update_notification_content_title))
-                    .setContentText(APP.getString(R.string.app_update_notification_content_text)
+                    .setContentTitle(application
+                            .getString(R.string.app_update_notification_content_title))
+                    .setContentText(application
+                            .getString(R.string.app_update_notification_content_text)
                             + " " + versionName);
 
             final NotificationManagerCompat notificationManager
-                    = NotificationManagerCompat.from(APP);
+                    = NotificationManagerCompat.from(application);
             notificationManager.notify(notificationId, notificationBuilder.build());
         }
     }
 
-    private boolean isConnected() {
-        final ConnectivityManager cm = ContextCompat.getSystemService(APP,
+    private static boolean isConnected(@NonNull final App app) {
+        final ConnectivityManager cm = ContextCompat.getSystemService(app,
                 ConnectivityManager.class);
         return cm.getActiveNetworkInfo() != null
                 && cm.getActiveNetworkInfo().isConnected();
     }
+
+    public static boolean isGithubApk(@NonNull final App app) {
+        return getCertificateSHA1Fingerprint(app).equals(GITHUB_APK_SHA1);
+    }
+
+    public static void checkNewVersion(@NonNull final App app) {
+        final SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(app);
+
+        // Check if user has enabled/disabled update checking
+        // and if the current apk is a github one or not.
+        if (!prefs.getBoolean(app.getString(R.string.update_app_key), true)
+                || !isGithubApk(app)) {
+            if (disposable != null) {
+                disposable.dispose();
+            }
+            return;
+        }
+
+        disposable = Observable.fromCallable(() -> {
```

Comment:
Instead of saving the disposable to a static variable, which would actually create a leak, return it and, in the `App` class save it as a field to be `dispose()`d in the `App`'s `onDestroy()`

Reply:
Right, my bad. Thanks!